cmake/utils.cmake

```diff
@@ -114,3 +116,25 @@ function(file_is_in_directory varname file dir)
         set(${varname} TRUE PARENT_SCOPE)
     endif()
 endfunction()
+
+#
+# Sets the `ENABLE_HARDENING_DEFAULT`, `HARDENING_FLAGS` and `HARDENING_LDFLAGS`
+# variables for passing hardening compiler flags.
+#
+# Uses `ENABLE_FUZZER` and `TARGET_OS_FREEBSD` variables.
+function(set_hardening_flags)
```

Comment:
I don't think this belongs in `utils.cmake`. Maybe better add a new file `hardening.cmake` for this.

Reply:
Fixed: moved the code to a new `cmake/hardening.cmake` module.